services/app-web/src/common-code/domain-models/submission.ts

```diff
@@ -135,6 +136,64 @@ function submissionName(
     return `MCR-${submission.stateCode.toUpperCase()}-${formattedProgramNames}-${padNumber}`
 }
 
+export type RateDataType = {
+    rateType?: 'AMENDMENT' | 'NEW' | null
+    rateDateStart?: Date
+    rateDateEnd?: Date
+    rateDateCertified?: Date
+    rateAmendmentInfo?: {
+        effectiveDateEnd?: Date
+        effectiveDateStart?: Date
+    } | null
+}
+
+export const generateRateName = (
+    rateData: Partial<RateDataType>,
```

Comment:
Why is this `Partial`? It seems like everything can already be null above. The question marks should make things optional.

Reply:
Yes, totally forgot to remove this when switching to `RateDataType`.